tests/Unit/API/Google/AdsCampaignTest.php

```diff
@@ -300,6 +312,45 @@ public function test_create_campaign_exception() {
 		}
 	}
 
+	public function test_create_campaign_exception_invalid_location_id() {
```

Comment:
> Missing a test either here or in the control, for a valid country code but not on our supported list. Ex: KP

I added the test of unsupported countries in the controller via e790283, thanks!

> Also I'm not sure if this test would have been better placed in AdsCampaignCriterionTest.php. 

This test is the case when we send invalid location IDs to Google Ads API via `mutate` method so I generated a mutate mock exception to mimic the situation. So I think it would be good to place it in `AdsCampaignTest`.

Reply:
Thanks makes sense, I agree it belongs more to the situation where we call the mutate. I just found it a shame that the tests didn't catch the error. Which made me wonder who has the responsibility of validating that location_ids is an array of only int's. And whether we should be enforcing that internally, or if we are content to rely on the validation in the controller.